### PR TITLE
[CN-563] Add unit tests for HotBackup Reconciler

### PR DESCRIPTION
**Pull request by mtyazici**
Created unit tests are taken from the deleted tests in  #381
Also added unit test for the case where one member's timed member state returns failure, this case is created for the fixed issue #445

### CODE REVIEW

File: controllers/hazelcast/hot_backup_controller_test.go [hunk 1]
```diff
@@ -13,31 +23,181 @@ import (
...
+		defer reconcileWg.Done()
+		_, _ = r.Reconcile(context.TODO(), reconcile.Request{NamespacedName: nn})
+	}()
+	restCallWg.Done()
+	reconcileWg.Wait()
+
+	Expect(hotBackupTriggers).Should(Equal(int32(1)))
+}
+
+func TestHotBackupReconciler_shouldCancelContextIfHazelcastCRIsDeleted(t *testing.T) {
```
Comment (cagric0) [inline controllers/hazelcast/hot_backup_controller_test.go:160]: Shouldn't this be .....If**HotbackupCR**IsDeleted?

**Review by cagric0 — COMMENTED** [review]
I've made a little refactoring. If you are not okay with it, I can revert it. Thus, just one naming issue other than that LGTM!

File: controllers/hazelcast/hot_backup_controller_test.go (hunk 1 above)
Comment (mtyazici) [reply controllers/hazelcast/hot_backup_controller_test.go:160]:
```suggestion
func TestHotBackupReconciler_shouldCancelContextIfHotbackupCRIsDeleted(t *testing.T) {
```

You are right

**Review by cagric0 — APPROVED** [review]

**Review by SeriyBg — APPROVED** [review]

**Comment by mtyazici** [comment]
Merged the PR since failure is unrelated to the PR.

**Pull request merged**